platforms/common/module.cpp

```diff
@@ -46,6 +46,168 @@
 
 pthread_mutex_t px4_modules_mutex = PTHREAD_MUTEX_INITIALIZER;
 
+BlockingList<ModuleBaseInterface *> _px4_modules_list;
```

Comment:
Why do you use `BlockingList`, as there's already a `px4_modules_mutex` mutex?

Reply:
I don't recall at this point, but I was probably trying to reuse another common piece until I fully realized what was needed. I'll take another pass.